internal/ethapi/api.go

```diff
@@ -1759,6 +1760,36 @@ func (s *TransactionAPI) sign(addr common.Address, tx *types.Transaction) (*type
 
 // SubmitTransaction is a helper function that submits tx to txPool and logs a message.
 func SubmitTransaction(ctx context.Context, b Backend, tx *types.Transaction) (common.Hash, error) {
+	// if we are running the flame node in auctioneer mode. we need to reject transactions being sent if the optimistic block
+	// height deviates from the soft block height by more than 1 block. The optimistic block is either one block ahead of the
+	// soft block or is at the same height as that of the soft block.
+	// If the soft block deviates from the optimistic block by more than 1 block, then we risk validating searcher transactions
+	// on older state since we do stateful validation of searcher txs on optimistic blocks. This is an undesirable situation and we
+	// ideally want to fix this deviation between optimistic blocks and soft blocks before allowing searchers to submit txs.
```

Comment:
In general the comments here feel a little lengthy and unclear... would be nice to shorten them and make it a bit more to the point.

I think something along the lines of:


> When running in auctioneer mode, we only want to accept txs when a submission has a chance of winning an auctioneer auction. We can't do this perfectly with information available, but we can block tx submission when optimistic block is clearly behind which would result in a submitted tx being dropped before next auction.